Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM jupyter/minimal-notebook
+LABEL maintainer="Hiromu Hota <hiromu.hota@hal.hitachi.com>"
+USER root
+
+RUN apt-get update && apt-get install -y \
+    libxml2-dev \
+    libxslt-dev \
+    python-matplotlib \
+    poppler-utils \
+    postgresql-client \
+    libmagickwand-dev \
+    ghostscript \
+ && rm -rf /var/lib/{apt,dpkg,cache,log}/
+RUN rm /etc/ImageMagick-6/policy.xml
+
+RUN pip install torch==0.4.1
+RUN pip install \
+    fonduer==0.3.4 \
+    matplotlib \
+    snorkel-metal==0.2.4
+
+RUN python -m spacy download en
+RUN conda install -y -c conda-forge ipywidgets
+
+USER $NB_UID
```

Comment:
Here's another example of a variable where I'm not sure where it comes from or how it is set. Will this work for all users?

Reply:
The direct answer to your question is no it won't work for all **linux** users within a Docker container, but this is how Docker works and should be ok for tutorials.
This Docker image makes it easy to jump start their Fonduer tutorials and is never meant to be used in production.
Well, end users have to have Docker installed but this is increasingly becoming popular.

Once a Docker container is deployed, the jupyter server is ran by `jovyan`.
As you know, jupyter itself is a web app and can serve multiple **human** users, which is good enough for tutorials.

Reply:
Having said, you don't want multiple human users to use a single Fonduer instance.
In practice, each user has to have a dedicated Fonduer instance for these tutorials, which could be deployed to their local PC or on the cloud.